### PR TITLE
HOSTEDCP-1019: Add create cluster for Agent for HCP CLI

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Add create cluster for Agent for HCP CLI

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1019](https://issues.redhat.com/browse/HOSTEDCP-1019)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: product-cli/cmd/cluster/agent/create.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		Short:        "Creates basic functional HostedCluster resources on Agent",
+		SilenceUsage: true,
+	}
+
+	opts.AgentPlatform = core.AgentPlatformCreateOptions{
+		APIServerAddress: "",
+		AgentNamespace:   "",
+	}
+
+	cmd.Flags().StringVar(&opts.AgentPlatform.APIServerAddress, "api-server-address", opts.AgentPlatform.APIServerAddress, "The API server address that should be used for components outside the control plane")
```
Comment (enxebre) [inline product-cli/cmd/cluster/agent/create.go:23]: Is there anything else we can add to make this more clear? what are components outside the control plane?

Comment (bryan-cox) [reply product-cli/cmd/cluster/agent/create.go:23]: @avishayt @eranco74 - is there anything additional here we can add?

Comment (eranco74) [reply product-cli/cmd/cluster/agent/create.go:23]: This is the hosted cluster kubeapi endpoint.   
Perhaps
```The API server address is the IP address for Kubernetes API communication.```

**Comment by bryan-cox** [comment]
CI was broken previously.
/test e2e-aws

**Pull request merged**